woocommerce/woocommerce-ios
# [Mobile Payments] Apply secondary style to cancel button in modal alerts

Fixes #4364

This switches the style of the cancel buttons to secondary on the alerts for scanning for readers, when reader is ready, and when it's waiting for a card to be tapped (which seems identical to the previous one).

For this, I introduced a new `.secondaryOnly` action mode to `CardPresentPaymentsModalViewModel`, in which there is only one secondary button and no primary. This feels a bit awkward, but it seemed like the simplest way forward without changing the modals too much.

I also added a bunch of new tests for `CardPresentPaymentsModalViewController` to make sure the visible buttons match what's expected for each action mode.

## To test

1. Without a reader connected, go to an order and tap Collect Payment
2. The alert should display the Cancel button in a secondary style.
3. Tapping Cancel should dismiss the modal
4. Connect to a reader.
5. The alert shown when the reader is ready to collect payment should show the Cancel button in a secondary style.
6. Tapping Cancel should cancel payment

Scanning for reader | Collecting payment
-|-
![Screen Shot 2021-06-04 at 13 32 24](https://user-images.githubusercontent.com/8739/120798146-38b51300-c53d-11eb-9124-93a93e7e5093.png)|![Screen Shot 2021-06-04 at 14 00 24](https://user-images.githubusercontent.com/8739/120798151-3a7ed680-c53d-11eb-8734-511e01a33d39.png)


Update release notes:
